drivers/gles3/rasterizer_gles3.cpp

```diff
@@ -298,54 +298,89 @@ void RasterizerGLES3::blit_render_targets_to_screen(DisplayServer::WindowID p_sc
-
-		} else {
-			//scale vertically
-			screenrect.size.x = win_size.width;
-			screenrect.size.y = imgrect.size.y * win_size.width / imgrect.size.x;
-			screenrect.position.y = (win_size.height - screenrect.size.y) / 2;
-		}
-	} else {
-		screenrect = imgrect;
-		screenrect.position += ((Size2(win_size.width, win_size.height) - screenrect.size) / 2.0).floor();
+	switch (p_stretch_mode) {
+		case RenderingServer::SPLASH_STRETCH_MODE_DISABLED: {
+			screenrect = imgrect;
+			screenrect.position += ((window_size - screenrect.size) / 2.0).floor();
+		} break;
+		case RenderingServer::SPLASH_STRETCH_MODE_KEEP: {
+			if (window_size.width > window_size.height) {
+				// Scale horizontally.
+				screenrect.size.y = window_size.height;
+				screenrect.size.x = imgrect.size.x * window_size.height / imgrect.size.y;
+				screenrect.position.x = (window_size.width - screenrect.size.x) / 2;
+			} else {
+				// Scale vertically.
+				screenrect.size.x = window_size.width;
+				screenrect.size.y = imgrect.size.y * window_size.width / imgrect.size.x;
+				screenrect.position.y = (window_size.height - screenrect.size.y) / 2;
+			}
+		} break;
+		case RenderingServer::SPLASH_STRETCH_MODE_KEEP_WIDTH: {
+			// Scale vertically.
+			screenrect.size.x = window_size.width;
+			screenrect.size.y = imgrect.size.y * window_size.width / imgrect.size.x;
+			screenrect.position.y = (window_size.height - screenrect.size.y) / 2;
+		} break;
+		case RenderingServer::SPLASH_STRETCH_MODE_KEEP_HEIGHT: {
+			// Scale horizontally.
+			screenrect.size.y = window_size.height;
+			screenrect.size.x = imgrect.size.x * window_size.height / imgrect.size.y;
+			screenrect.position.x = (window_size.width - screenrect.size.x) / 2;
+		} break;
+		case RenderingServer::SPLASH_STRETCH_MODE_COVER: {
+			double window_aspect = (double)window_size.width / window_size.height;
+			double img_aspect = imgrect.size.x / imgrect.size.y;
+
+			if (window_aspect > img_aspect) {
+				// Scale vertically.
+				screenrect.size.x = window_size.width;
+				screenrect.size.y = imgrect.size.y * window_size.width / imgrect.size.x;
+				screenrect.position.y = (window_size.height - screenrect.size.y) / 2;
+			} else {
+				// Scale horizontally.
+				screenrect.size.y = window_size.height;
+				screenrect.size.x = imgrect.size.x * window_size.height / imgrect.size.y;
+				screenrect.position.x = (window_size.width - screenrect.size.x) / 2;
+			}
+		} break;
+		case RenderingServer::SPLASH_STRETCH_MODE_EXPAND: {
+			screenrect.size.x = window_size.width;
+			screenrect.size.y = window_size.height;
+		} break;
 	}
 
+	// FIXME: Actually draw the image after binding it, using screenrect for scaling.
+
 	RasterizerStorageGLES3::Texture *t = storage.texture_owner.get_or_null(texture);
 	glActiveTexture(GL_TEXTURE0 + storage.config.max_texture_image_units - 1);
 	glBindTexture(GL_TEXTURE_2D, t->tex_id);
+	//canvas->draw_generic_textured_rect(screenrect, Rect2(0, 0, 1, 1));
```

Comment:
Converting this to 4.0 GLES3 code is missing (that's the equivalent code from `3.x`), and that's why no boot splash gets rendered at all with OpenGL3.